rakelib/plugins_docs_dependencies.rake

```diff
@@ -205,6 +204,14 @@ task :generate_plugins_version do
         end
       end
     end
+    DepProxy.class_eval do
```

Comment:
🤔 It looks like `Bundler::DepProxy` is [completely removed upstream in Bundler 2.4.0](https://github.com/rubygems/rubygems/commit/eb697fb827db8e280f2b43bc06e4cc0aa3dc1bbd).

A possible way to handle this with the as-resolved-in-jruby-9.2 and also in the upcoming 9.4 would be to define this as a stand-alone helper method that logs a warning when it is no longer needed:

~~~ ruby
def reset_bundler_dep_proxy!
  if defined?(Bundler::DepProxy) && Bundler::DepProxy.instance_variables.include?(:@proxies)
    Bundler::DepProxy.instance_variable_get(:@proxies).clear
  else
    warn("Bundler::DepProxy could not be reset! The patch may no longer be necessary, or may need to be modified.")
  end
end
~~~

Reply:
I don't think we this sort of runtime detection since jruby won't be configurable or upgradable. After this PR is merged to main we can reintroduce the changes to accomodate bundler 2.4 done in https://github.com/elastic/logstash/pull/14995/files#diff-a2ab6175e1d1e05f94696fb680f4454320c3d03777d8d73d7e9d3c8a375acd80L207